tests/integration/conftest.py

```diff
@@ -22,6 +25,35 @@
 session = requests.session()
 
 
+@pytest.hookimpl
+def pytest_sessionstart(session):
+    no_fds_soft, no_fds_hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+    if no_fds_soft < 2028:
+        session.warn(
+            pytest.PytestWarning(
+                f"Too low open filedescriptor limit: {no_fds_soft} (try `ulimit -n 4096`)"
+            )
+        )
```

Comment:
This warning is a bit pants and very easily missed, I looks like this:
```
> pytest tests/integration/test_basic.py -k malformed_objects --collect-only                                                                                                           1s 400ms
/Users/flub/code/symbolicator:0: PytestWarning: Too low open filedescriptor limit: 400 (try `ulimit -n 4096`)
===================================================================================== test session starts =====================================================================================
platform darwin -- Python 3.8.2, pytest-5.4.3, py-1.8.1, pluggy-0.13.1
rootdir: /Users/flub/code/symbolicator, inifile: pytest.ini
plugins: rerunfailures-9.0, localserver-0.5.0, xdist-1.33.0, forked-1.1.3, icdiff-0.5
collected 30 items / 29 deselected / 1 selected
<Module tests/integration/test_basic.py>
  <Function test_malformed_objects>
```

Perhaps calling pytest.exit() is better, but it's very rude:
```
> pytest tests/integration/test_basic.py -k malformed_objects                                                                                                                             581ms
Exit: Too low open filedescriptor limit: 400 (try `ulimit -n 4096`)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! _pytest.outcomes.Exit: Too low open filedescriptor limit: 400 (try `ulimit -n 4096`) !!!!!!!!!!!!
```

If we prefer this perhaps we need an command line option to ignore the check